services/QuillLMS/app/controllers/students_controller.rb

```diff
@@ -107,6 +107,7 @@ def join_classroom
     return redirect_to profile_path unless current_user&.student?
 
     if classroom_id && unit_id
+      flash_closed_unit_error
```

Comment:
Do we need to remove the line below this that calls `flash_missing_unit_error`?  That one looks like it'd overwrite the error we want here?

Reply:
No - they are different errors, and since `closed_unit` redirects, that line will never be called if they are supposed to see the closed unit error.